packages/aila/src/core/prompt/builders/AilaLessonPromptBuilder.ts

```diff
@@ -77,16 +77,11 @@ export class AilaLessonPromptBuilder extends AilaPromptBuilder {
     baseLessonPlan: LooseLessonPlan | undefined,
   ): string {
     const lessonPlan = this._aila?.lessonPlan ?? {};
-    const { title, subject, keyStage, topic } = lessonPlan;
     const args: TemplateProps = {
-      subject,
-      lessonTitle: title,
-      keyStage,
-      topic,
-      currentLessonPlan: compressedLessonPlanForRag(lessonPlan),
+      lessonPlan,
       relevantLessonPlans,
       summaries: "None",
-      responseMode: "interactive",
+      responseMode: this._aila?.options.mode ?? "interactive",
```

Comment:
Allow us to use "generate" mode

Reply:
We should check the database for the lessonPlans table - might be blank for new rows?